crates/next-core/src/next_font_google/mod.rs

```diff
@@ -205,9 +221,61 @@ async fn get_stylesheet_url_from_request(request_json: StringVc) -> Result<Strin
 #[turbo_tasks::value(transparent)]
 struct NextFontGoogleOptions(self::options::NextFontGoogleOptions);
 
+#[turbo_tasks::value(transparent)]
+struct FontCssProperties {
+    font_family: StringVc,
+    weight: OptionU16Vc,
+    style: OptionStringVc,
+}
+
 #[turbo_tasks::function]
-async fn options_from_request(request: StringVc) -> Result<NextFontGoogleOptionsVc> {
-    let request: NextFontRequest = serde_json::from_str(&request.await?)?;
+async fn get_font_css_properties(options: NextFontGoogleOptionsVc) -> Result<FontCssPropertiesVc> {
+    let options = &*options.await?;
+
+    let font_family = format!(
+        "'{}'{}",
+        options.font_family,
+        options
+            .fallback
+            .as_ref()
+            .map(|fallback| {
+                ", ".to_owned()
+                    + &fallback
+                        .iter()
+                        .map(|f| format!("'{}'", f))
+                        .collect::<Vec<String>>()
+                        .join(", ")
+            })
+            .unwrap_or_else(|| "".to_owned())
+    );
```

Comment:
Nit: this looks like a vec join, but it's written in a way that concats a start and a rest. Given we're already allocating a second vec, can we just make a full vec and join a single time?